src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -4454,12 +4524,89 @@ private void GenerateOneCode()
+                            RegexCharClass.IsNegated(set!))
+                        {
+                            // Similarly, if this is a setloop{atomic} and we're left-to-right and case-sensitive,
+                            // and if the set contains only 2 or 3 negated chars, we can use the vectorized IndexOfAny
+                            // to search for those chars.
+
+                            // i = runtext.AsSpan(runtextpos, len).IndexOfAny(ch1, ch2{, ch3});
+                            Ldloc(_runtextLocal!);
+                            Ldloc(_runtextposLocal!);
+                            Ldloc(lenLocal);
+                            Call(s_stringAsSpanIntIntMethod);
+                            Ldc(setChars[0]);
+                            Ldc(setChars[1]);
+                            if (numSetChars == 2)
+                            {
+                                Call(s_spanIndexOfAnyCharChar);
+                            }
+                            else
+                            {
+                                Ldc(setChars[2]);
+                                Call(s_spanIndexOfAnyCharCharChar);
+                            }
+                            Stloc(iLocal);
+
+                            Label charFound = DefineLabel();
+
+                            // if (i != -1) goto charFound;
+                            Ldloc(iLocal);
+                            Ldc(-1);
+                            Bne(charFound);
+
+                            // runtextpos += len;
+                            // i = 0;
+                            // goto loopEnd;
+                            Ldloc(_runtextposLocal!);
+                            Ldloc(lenLocal);
+                            Add();
+                            Stloc(_runtextposLocal!);
+                            Ldc(0);
+                            Stloc(iLocal);
+                            BrFar(loopEnd);
+
+                            // charFound:
+                            // runtextpos += i;
+                            // i = len - i;
+                            // goto loopEnd;
+                            MarkLabel(charFound);
+                            Ldloc(_runtextposLocal!);
+                            Ldloc(iLocal);
+                            Add();
+                            Stloc(_runtextposLocal!);
+                            Ldloc(lenLocal);
+                            Ldloc(iLocal);
+                            Sub();
+                            Stloc(iLocal);
+                            BrFar(loopEnd);
+                        }
+                        else if ((Code() == RegexCode.Setloop || Code() == RegexCode.Setloopatomic) &&
+                            !IsRightToLeft() &&
+                            set == RegexCharClass.AnyClass)
+                        {
+                            // If someone uses .* along with RegexOptions.Singleline, that becomes [anycharacter]*, which means it'll
```

Comment:
> This will jump to the end of the text, but then it will start back tracking?

Yes, that's how it already works... this just does it faster.

> Oh, this is only if it's atomic?

Nope. Only if it's greedy, whether atomic or not.